cumulusci/core/keychain/tests/test_encrypted_file_project_keychain.py

```diff
@@ -188,17 +255,117 @@ def test_load_service_files(self, keychain):
+        assert built_in_connected_app is DEFAULT_CONNECTED_APP
+
+    def test_get_service__with_class_path(self, keychain, service_config):
+        encrypted = keychain._encrypt_config(service_config)
+        keychain.config["services"]["marketing_cloud"] = {"foo": encrypted}
+        mc_service = keychain._get_service("marketing_cloud", "foo")
+
+        assert isinstance(mc_service, MarketingCloudServiceConfig)
 
+    @mock.patch("cumulusci.core.keychain.encrypted_file_project_keychain.import_class")
+    def test_get_service__bad_class_path(self, import_class, keychain, service_config):
+        import_class.side_effect = AttributeError
+        encrypted = keychain._encrypt_config(service_config)
+        keychain.config["services"]["marketing_cloud"] = {"foo": encrypted}
+
+        error_message = "Unrecognized class_path for service: cumulusci.core.config.marketing_cloud_service_config.MarketingCloudServiceConfig"
+        with pytest.raises(CumulusCIException, match=error_message):
+            keychain._get_service("marketing_cloud", "foo")
+
+    def test_get_service__does_not_exist(self, keychain, service_config):
+        keychain.set_service("github", "alias", service_config)
+        error_message = "No service of type github exists with the name: does-not-exist"
+        with pytest.raises(ServiceNotConfigured, match=error_message):
+            keychain.get_service("github", "does-not-exist")
+
+    @pytest.mark.skipif(
+        sys.platform.startswith("win"),
+        reason="Windows has a different value returned by os.stat",
+    )
     def test_set_service_github(self, keychain, service_config):
         keychain.set_service("github", "alias", service_config)
         default_github_service = keychain.get_service("github")
 
+        service_filepath = Path(
+            keychain.global_config_dir, "services/github/alias.service"
+        )
+        assert service_filepath.is_file()
+
+        # ensure expected file permissions
+        stat_result = os.stat(service_filepath)
+        actual_mode = oct(stat_result.st_mode & 0o777)
+        assert actual_mode == oct(SERVICE_ORG_FILE_MODE)
+
         assert default_github_service.config == {
             **service_config.config,
             "token": "test123",
         }
 
+    @pytest.mark.skipif(
+        not sys.platform.startswith("win"),
+        reason="Windows has a different value returned by os.stat",
+    )
+    def test_set_service_github__windows_has_correct_file_perms(
+        self, keychain, service_config
+    ):
+        keychain.set_service("github", "alias", service_config)
+
+        service_filepath = Path(
+            keychain.global_config_dir, "services/github/alias.service"
+        )
+        assert service_filepath.is_file()
+
+        # ensure expected file permissions
+        stat_result = os.stat(service_filepath)
+        actual_mode = oct(stat_result.st_mode & 0o777)
+        assert actual_mode == "0o666"
```

Comment:
From what I can gather on inspection of the file in windows, it seems to have the correct file permissions. I've verified with `cmd.exe` using [icacls](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/icacls#additional-references) which _seems_ like what we want:
```
(cci-venv) D:\Users\bparker\cumulusci>icacls \Users\bparker\AppData\Local\Temp\1\fake_home_tj8bb3in\.cumulusci\services\github\alias.service
\Users\bparker\AppData\Local\Temp\1\fake_home_tj8bb3in\.cumulusci\services\github\alias.service NT AUTHORITY\SYSTEM:(I)(F)
                                                                                                BUILTIN\Administrators:(I)(F)
                                                                                                roundata\bparker:(I)(F)

Successfully processed 1 files; Failed processing 0 files
```
This seems to indicate that there are three "groups" of users that have the `(I)` and `(F)`:
* NT AUTHORITY\SYSTEM
* BUILTIN\Administrators
* roundata\bparker

The permissions are as follows:
* (I) - permission inherited from the parent container
* (F) - full access

I can't find anything that mentions why `os.stat()` would return a value that differs on windows environments.

Reply:
I think it's just that os.stat wraps a system call that does something different on Windows.

Thanks for checking the permissions. I'm fine accepting that and not testing the permissions on Windows if there isn't a straightforward way to do so, but I'd rather not skip the rest of the assertions in the same tests.